### PR TITLE
chore: remove deprecation warnings for pmd-7.0

### PR DESCRIPTION
**Pull request by soloturn**
inline the rules, without change. adapt the ones moved into another one, remove one not any more supported. these actions have dedicated commits to make it easer to follow.

### CODE REVIEW

**Comment by soloturn** [comment]
here info on how to convert:
https://docs.pmd-code.org/latest/pmd_release_notes_pmd7.html

**Comment by soloturn** [comment]
@jdrueckert any interest to merge this one, and publishing a new release?

File: pmd/pmd.xml [hunk 1]
```diff
@@ -7,8 +7,41 @@
...
+  <rule ref="category/java/performance.xml/BooleanInstantiation" deprecated="true" />
+
+  <rule ref="category/java/design.xml/CollapsibleIfStatements" deprecated="true" />
+  <rule ref="category/java/design.xml/SimplifiedTernary" deprecated="true" />
+<!--  inline <rule ref="rulesets/java/braces.xml"/>-->
+  <rule ref="category/java/codestyle.xml/ForLoopsMustUseBraces" deprecated="true" />
+  <rule ref="category/java/codestyle.xml/IfElseStmtsMustUseBraces" deprecated="true" />
+  <rule ref="category/java/codestyle.xml/IfStmtsMustUseBraces" deprecated="true" />
+  <rule ref="category/java/codestyle.xml/WhileLoopsMustUseBraces" deprecated="true" />
   <rule ref="rulesets/java/clone.xml"/>
```
Comment (jdrueckert) [inline pmd/pmd.xml]: would it make sense to have some kind of "end of inline" comment?

File: pmd/pmd.xml [hunk 2]
```diff
@@ -8,46 +8,46 @@
...
+  <rule ref="category/java/codestyle.xml/IfStmtsMustUseBraces" />
+  <rule ref="category/java/codestyle.xml/WhileLoopsMustUseBraces" />
+<!--  <rule ref="rulesets/java/clone.xml"/>-->
+<!--  <rule ref="rulesets/java/finalizers.xml"/>-->
+<!--  <rule ref="rulesets/java/logging-java.xml">-->
+<!--      <exclude name="GuardLogStatementJavaUtil"/>-->
+<!--  </rule>-->
+<!--  <rule ref="rulesets/java/migrating.xml"/>-->
+<!--  <rule ref="rulesets/java/sunsecure.xml"/>-->
+<!--  <rule ref="rulesets/java/unusedcode.xml"/>-->
```
Comment (jdrueckert) [inline pmd/pmd.xml]: why are these commented out now?

File: pmd/pmd.xml [hunk 3]
```diff
@@ -8,46 +8,46 @@
   </description>
   <exclude-pattern>.*/org.terasology.protobuf/.*</exclude-pattern>
 <!-- inline <rule ref="rulesets/java/basic.xml"/>  -->
-  <rule ref="category/java/errorprone.xml/AvoidBranchingStatementAsLastInLoop" deprecated="true" />
```
Comment (jdrueckert) [inline pmd/pmd.xml]: why can we just remove the deprecation note for this and the others?
was the deprecation related to not having them inlined? :thinking:

File: pmd/pmd.xml [hunk 4]
```diff
@@ -56,7 +56,6 @@
 <!--      <exclude name="GuardLogStatementJavaUtil"/>-->
 <!--  </rule>-->
   <rule ref="category/java/errorprone.xml/InvalidSlf4jMessageFormat" />
-  <rule ref="category/java/errorprone.xml/LoggerIsNotStaticFinal" />
```
Comment (jdrueckert) [inline pmd/pmd.xml]: https://docs.pmd-code.org/latest/pmd_release_notes_pmd7.html says to instead use `ProperLogger`

**Review by jdrueckert — COMMENTED** [review]
@soloturn thank you for the contribution and reminding me about it, it slipped my radar...
I left a bunch of comments, please have a look :slightly_smiling_face:

File: pmd/pmd.xml (hunk 3 above)
Comment (soloturn) [reply pmd/pmd.xml]: yes, mainly the way rulesets were organized changed, not the rules itself. it looks like their way of deprecating the ruleset was to mark the containing rules as deprecated. take this example AvoidBranchingStatementAsLastInLoop, here NOT marked as deprecated: https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#avoidbranchingstatementaslastinloop.

File: pmd/pmd.xml (hunk 1 above)
Comment (soloturn) [reply pmd/pmd.xml]: all is inlined and stems from rulesets. start of next ruleset is end of inline. and the comments will most likely go away, i just left them to make this review easier.

File: pmd/pmd.xml (hunk 2 above)
Comment (soloturn) [reply pmd/pmd.xml]: inlined as well. i added a comment to make it more clear. all is inlined.

File: pmd/pmd.xml (hunk 3 above)
Comment (soloturn) [reply pmd/pmd.xml]: @jdrueckert anything missing or it can be merged as is?

**Comment by soloturn** [comment]
@skaldarnar you d be able to do this as well as long @jdrueckert is busy?

**Comment by soloturn** [comment]
@BenjaminAmos, am i allowed to ping you here as well?

**Comment by jdrueckert** [comment]
@soloturn apologies for the delay... been attempting to go on vacation last week but fell ill and needed a few days to get back home and recover. also, we need to include TeraConfig in the activity feed 🙄 
should be fine to go, thank you for being persistent 💚

**Review by jdrueckert — APPROVED** [review]

**Comment by soloturn** [comment]
@jdrueckert @BenjaminAmos @skaldarnar can you please release it, so it can be included into the other projects?

**Comment by soloturn** [comment]
or, @keturn you know how to make it available?

**Pull request merged**